src/Extensions/TimedScopes/Internal/TimedScopeEventSender.cs

```diff
@@ -0,0 +1,97 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT license.
+
+using System;
+using System.Diagnostics;
+using System.Globalization;
+using Microsoft.Extensions.Hosting;
+using Microsoft.Extensions.Logging;
+using Microsoft.Omex.Extensions.Abstractions;
+
+namespace Microsoft.Omex.Extensions.TimedScopes
+{
+	internal sealed class TimedScopeEventSender : ITimedScopeEventSender
+	{
+		public TimedScopeEventSender(TimedScopeEventSource eventSource, IHostEnvironment hostEnvironment, ILogger<TimedScopeEventSender> logger)
+		{
+			m_eventSource = eventSource;
+			m_serviceName = hostEnvironment.ApplicationName;
+			m_logger = logger;
+		}
+
+
+		public void LogTimedScopeEndEvent(TimedScope scope)
+		{
+			if (!m_eventSource.IsEnabled())
+			{
+				return;
+			}
+
+			string serviceName = m_serviceName;
+			string subtype = scope.SubType;
+			string metadata = scope.Metadata;
+			TimedScopeResult result = scope.Result;
+			Activity activity = scope.Activity;
+			string name = activity.OperationName;
+			string correlationId = activity.Id;
+			double durationMs = activity.Duration.TotalMilliseconds;
+			string userHash = activity.GetUserHash(); //Breaking Change: field not set
+			bool isTransaction = activity.IsTransaction(); //Breaking Change: field not set
+
+			string nameAsString = SanitizeString(name, nameof(name), name);
+			string subTypeAsString = SanitizeString(subtype, nameof(subtype), name);
+			string metaDataAsString = SanitizeString(metadata, nameof(metadata), name);
+			string userHashAsString = SanitizeString(userHash, nameof(userHash), name);
+			string serviceNameAsString = SanitizeString(serviceName, nameof(serviceName), name);
+			string correlationIdAsString = SanitizeString(correlationId, nameof(correlationId), name);
+			string resultAsString = result.ToString();
+			long durationMsAsLong = Convert.ToInt64(durationMs, CultureInfo.InvariantCulture);
+
+			if (isTransaction)
+			{
+				m_eventSource.WriteTimedScopeTestEvent(
+					nameAsString,
+					subTypeAsString,
+					metaDataAsString,
+					serviceNameAsString,
+					resultAsString,
+					correlationIdAsString,
+					durationMsAsLong);
+			}
+			else
+			{
+				m_eventSource.WriteTimedScopeEvent(
+					nameAsString,
+					subTypeAsString,
+					metaDataAsString,
+					userHashAsString,
+					serviceNameAsString,
+					resultAsString,
+					correlationIdAsString,
+					durationMsAsLong);
+			}
+		}
+
+
+		private string SanitizeString(string value, string name, string activityName)
```

Comment:
I don't think it would be beneficial here, because we need to return a string (since it's required by ETW) and we will be creating a new string from Span on every run, but in the case with string we will just call `Substing` only when string is too long, besides that there might be a possibility that new implementation of substing is build on Span if it's more beneficial.